feat: add new flag for astro check (#8838)
withastro/docs: src/content/docs/en/reference/cli-reference.mdx

Specifies not to clear the ouput between checks when in watch mode.

#### `--noSync`

> Specifies if the command should not run `astro sync` before checking the project.

Reviewer: ```suggestion
Specifies not to run `astro sync` before checking the project.
```
